pandas/core/arrays/datetimes.py

```diff
@@ -1644,6 +1644,8 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
         copy = False
     elif isinstance(data, ABCSeries):
         data = data._values
+    elif isinstance(data, ABCPandasArray):
+        data = data._ndarray
```

Comment:
This got added as a kludge to try to get tests passing, honestly not sure if its entirely necessary.

Reply:
Is the root problem the return type of Series[datetime64[ns]].array? I've been struggling with this as well.

That can return either a

1. PandasArray[datetime64[ns]]
2. DatetimeArray

Returning a DatetimeArray is *probably* more useful, and more consistent with the tz-aware case. However, it's also a lie because for tz-naive data `DatetimeArray.dtype` isn't an extension dtype. It's just a NumPy dtype. I worry that returning a DatetimeArray will bite us down the road when we go to do extension-dtypey things to `<thing>.array.dtype`.

I'm going to copy this over to the main PR so that it doesn't get lost.